From 875e8c0c40188689c87e936b9fee1dbe3480a9b1 Mon Sep 17 00:00:00 2001
From: RoEdAl <roed@onet.eu>
Date: Wed, 31 Aug 2022 10:38:35 +0200
Subject: [PATCH 09/12] wxWebViewEdge: handle more error codes

* COREWEBVIEW2_WEB_ERROR_STATUS_VALID_AUTHENTICATION_CREDENTIALS_REQUIRED
* COREWEBVIEW2_WEB_ERROR_STATUS_VALID_PROXY_AUTHENTICATION_REQUIRED.
---
 src/msw/webview_edge.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/msw/webview_edge.cpp b/src/msw/webview_edge.cpp
index 5302691..3198200 100644
--- a/src/msw/webview_edge.cpp
+++ b/src/msw/webview_edge.cpp
@@ -257,6 +257,8 @@ HRESULT wxWebViewEdgeImpl::OnNavigationCompleted(ICoreWebView2* WXUNUSED(sender)
                 WX_ERROR2_CASE(COREWEBVIEW2_WEB_ERROR_STATUS_HOST_NAME_NOT_RESOLVED, wxWEBVIEW_NAV_ERR_CONNECTION)
                 WX_ERROR2_CASE(COREWEBVIEW2_WEB_ERROR_STATUS_REDIRECT_FAILED, wxWEBVIEW_NAV_ERR_OTHER)
                 WX_ERROR2_CASE(COREWEBVIEW2_WEB_ERROR_STATUS_UNEXPECTED_ERROR, wxWEBVIEW_NAV_ERR_OTHER)
+                WX_ERROR2_CASE(COREWEBVIEW2_WEB_ERROR_STATUS_VALID_AUTHENTICATION_CREDENTIALS_REQUIRED, wxWEBVIEW_NAV_ERR_AUTH)
+                WX_ERROR2_CASE(COREWEBVIEW2_WEB_ERROR_STATUS_VALID_PROXY_AUTHENTICATION_REQUIRED, wxWEBVIEW_NAV_ERR_AUTH)
             case COREWEBVIEW2_WEB_ERROR_STATUS_OPERATION_CANCELED:
                 // This status is triggered by vetoing a wxEVT_WEBVIEW_NAVIGATING event
                 ignoreStatus = true;
-- 
2.36.1.windows.1

